### DISCUSSION TITLE
pcb for lighter board

### DISCUSSION THREAD

**Discussion by micked99** [mikaelnousiainen/RS41ng#27] [General]
I made a pcb with only the STM, 5351 and a Ublox Max gps on that should run this fw, its weighs less then 1,5 gram fully assembled, but, cant get it to work, anyone interested that think they can get it runing Im happy to send an assembled board or a bare pcb. 

I have uploaded all files here:  https://github.com/micked99/RS41ng_pcb

/Mikael

mikael_dagman@hotmail.com

**Comment by mikaelnousiainen** [comment]
@micked99 Very interesting! Any luck getting it running?

**Comment by micked99** [comment]
Hi, no unfortunately I havent really had the time to sit down to try a few thing, if you would like have to have a go I can send you an assembled board 👍 , if so just mail me on the above address.

/Mikael

**Comment by micked99** [comment]
I have uploaded the code where I have stripped most of the 4032 code from the fw, it running fine on the rs41 sonde with the added 5351 board but not on my custom pcb, any chance you can have a look at it ?

https://github.com/micked99/RS41ng_pcb/blob/main/RS41ng.rar

/Mikael

**Comment by Felix-S-IRS** [comment]
Hi,

electrical engineer here. I had a quick look at your schematic and found some problems:

- Your 24MHz quarz (i assume its a ABM8G-24.000MHZ-18-D2Y-T) lacks load capacitors. Therefore it might not even oscillate. Have you checked if it oscillates with an oscilloscope? Read this to get a better idea: https://www.iqdfrequencyproducts.com/blog/2020/08/03/myths-around-load-capacitance-how-to-choose-the-right-capacitors/
- I haven't looked into the datasheet of the microcontroller, clock generator and GPS yet, but i don't think it's wise to directly connect it to the battery voltage. You'll probably need some kind of (switching) voltage regulator there and giving them a stable supply voltage (for example 3.3V). Having a buck-boost converter will additionally allow your circuit to really suck the batteries dry and get everything out of them when their voltage drops towards the end of their life.

I will have a closer look later and make a revised design/recomendations how to botch your existing pcb.

Best regards
Felix

**Reply by Felix-S-IRS** [reply]
Are the pictures of the board the current state? Because the !reset resistor R3 is not mounted. This causes the microcontroller to be permanently in reset. Solder a 10K or something resistor there if you haven't already. (Also C2 and C3 while you're at it. Bypass capacitors help)

**Reply by Felix-S-IRS** [reply]
Ok, I've had a closer look at it and found several problems:
-Missing load capacitors on microcontroller 24 MHz crystal
-R3 not mounted -> microcontroller constantly in reset (if not pulled high by something else line the STLink)
-VCC-IO of GPS must be connected to VCC, otherwise its constantly in reset. RESET Pin should also be tied high for the same reason (altough it propably has an internal pullup, but it's not clearly stated in the datasheet)
-crystal-oscillator of the SI5351a (full oscillator circuit used instead of just the crystal and two load capacitors as recommended in the datasheet). This might work, but i don't know enough about the input of the SI5351a to judge that. Would recommend a normal crystal (with load capacitors!)
- Place more bypass capacitors
- Input supply not regulated. Would recommend integrated switchmode buck/boost converter to produce a stable 3.3V supply from a wide input voltage range.

Can you please specify in which ways it doesn't work?

- Can the microcontrolle be programmed/flashed with the STLink? (if not -> place load capacitors (about 18pf each. might be able to botch that) and R3)
- Is the GPS doing anything? (No -> connect VCC_IO and RESET to VIN. Can be done with botch wires)
- Everything above working but SI5351a does not produce RF -> change crystal oscilator for normal crystal (also needs load capacitors and changes in connections)

All in all I would recomend making a new PCB. I am changing the design a bit and will upload an improved version soon. For that i have a few questions:

- What kind of power supply should this pcb be connected to? Regulated 3.3V (recommended), directly to batteries (what kind), batteries with a solar charger or only a solar panel?
- Is the SI5151a schematic with the crystal oscillator instead of the normal crystal based on an existing working circuit? 
- Whats your prefered vendor for components?
- What temperature range are you aiming for? -40°C to 85°C?
- What are the exact names of the 24MHz crystal and the 25MHz oscillator?

Best regards and rtfm
Felix

**Reply by micked99** [reply]
Hi, tnx for looking into this :)

You are correct there is no caps on the 24Mhz xo but the stray capacistance of the board is more then enough to have it running, it programs just fine, better would be to have a 3 pin resonator but the idea is to lift the components of a rs41 board and mount them onto this pcb, thats why I went with this config, on this Im 100% theres no problem with the stm running.

As you state most micro controllers have a internal pullup of the rest pin, I have mostly worked with Atmega328 and on those board there is no need for pullup, as I did not know for sure on the stm I inserted the reset pullup resistor on the pcb just to be sure, I have soldered it on the board (10k) but still not firing up, I need to enable debug to see what it is spitting out.

It programs just fine with the st link, I see the current go up to about 58-60mA when I connect  power and then 30-40 sec later when the gps locks in the current drops of about 5-8 mA.

I made the pcb to hold several gps modules like Ublox Max and atgm336, for the ublox to work you are correct it need the vcc io to run so I have just solder bridged all four pins  , reset, vcc, vcc-io, vcc backup, also removed the bridge on the reset pin but it still works, guess weak internal pullup on this pin as well.

The 5351 does no tx any kind of rf, it uses a 25Mhz tcxo which I hear on my radio and see on the sdr if I place my thumb on it.

For my use I would only fly this with solar power, no batteries 6x cells in series usually works fine, the stm is a bit cranky about unstable solar in the morning , on the modified rs41 solar only flights I tried as a reference inserted a reset supervisor to keep it in reset until vcc was reasonable stable, iirc I used MAX809 2,93V reset voltage

yes I use regulated 3,3 from the st-link when testing on the bench, also a stabilized PS to be sure no ripple was introduced from the st-link.

I have made lots of long duration flights (more then 100) and have lots of components for those things laying around, mostly buy from Aliexpress and local firms.

working temp is mostly around -25 - -45°c

for tcxo look fro dsb321 or dsb211

Im quite sure the problem lies in the fw rather then the hw, but Im a hw guy and not that into programming, Im thinking, is there is check procedure during boot where the stm is looking for the uhf radio 4032? when it does not see it  it halts, anyway, I need to enable debug to see whats happening but those things really hearts my brain, lol :) 

/Mikael

**Comment by micked99** [comment]
Some more notes...

The tcxo for the 5351 needs to be a low ppm version, only 0,5ppm will work, anything else will drift of like crazy, most optimal would be to retrofit the 26Mhz tcxo on the rs41 which drives the gps and 4032 radio, this is a 0,5ppm module and have proven stable in really cold (-40°c) environment.

**Reply by Felix-S-IRS** [reply]
Ok if the problem doesn`t seem to be in the hw, I propably can't help much. Maybe I´ll build a similar board in the future, now that I have thought so much about it. Or at least add a Si5351 to a RS41.

**Reply by micked99** [reply]
Im happy to send you a bare board if you like, just let me know, you can mail me your address and it will be on its way :)

**Reply by Felix-S-IRS** [reply]
Thanks, but it would probably never go on its intended mission (unless we accidentally built a floater). I'm using RS41s as backup trackers for sciene payload carrying balloons for a student associations (KSat Stuttgart). All our flights are only a few hours. I have ordered a Si5351 breadboard for some tests. WSPR will only be a gimick for us. Our main balloon bus sends Lora for the telemetry downlink and the reprogrammed RS41 sending RTTY and Horus MFSK on 70cm is our more reliable backup.

https://www.ksat-stuttgart.de/en/our-projects/bubble/
If you want, you can send a board to our address, but it would probably never go on its grand journey only on small hops.

KSat e.V. Stuttgart
Institut für Raumfahrtsysteme - Universität Stuttgart
Pfaffenwaldring 29
70569 Stuttgart
Gemany

**Reply by whallmann** [reply]
Felix, did you got this si5351 running on shortwaves? I tried a small breakout board from Ali  on a rs41 but all i see is a very big noise hill on my kiwisdr with full display zoom-out 0-30 MHz if the tx fires. Also set all other modes off except horus 4fsk. tried 30m and 80m Band.
cu Wolf
